vendor/engines/saas/app/controllers/saas/coupons_controller.rb

```diff
@@ -11,8 +11,8 @@ def valid
     def redeem
       query = Queries::CouchCustomer.get_cust(params[:id], couch)
       doc = QueryHandler.exec(&query)
-      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
       plan_doc = doc
+      return render json: {success: false, message: "Couldn't find couch record: #{plan_doc.id}"} unless doc
```

Comment:
`plan_doc` is `false` if doc fails `unless` -

+1 to "To be more constructive, it seems like the error message needs to reference params[:id]" from @dahlbyk